services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java

```diff
@@ -69,6 +72,18 @@ public Upload(
     this.uploadKeyService = uploadKeyService;
   }
 
+  private List<DiagnosisKey> executeFederationUpload(UploadPayload payload) {
```

Comment:
suggest to use executeUploadAndCollectErrors(). as a method name to make it clear that the returned list is a collection of rejected keys.
Or maybe split the extraction of rejected keys in a separate method and just return the BatchUploadResponse structure from this method

Reply:
I think if we keep the original keys sorted by the same criteria (keyData) we can just collect the indexes from the upload response, and pass those to the markSuccesful...() method as a parameter. Something like ignoreIndexes.